src/test/java/com/google/cloud/spanner/pgadapter/python/PythonTransactionTests.java

```diff
@@ -0,0 +1,1095 @@
+    Assert.assertEquals(6, mockSpanner.countRequestsOfType(ExecuteSqlRequest.class));
+
+    List<ExecuteSqlRequest> requests = mockSpanner.getRequestsOfType(ExecuteSqlRequest.class);
+    List<String> sql = new ArrayList<>();
+
+    for (ExecuteSqlRequest request : requests) {
+      sql.add(request.getSql());
+    }
+
+    Assert.assertEquals(true, sql.contains(sql1));
+    Assert.assertEquals(true, sql.contains(sql2));
+    Assert.assertEquals(true, sql.contains(sql3));
+    Assert.assertEquals(true, sql.contains(sql4));
+    Assert.assertEquals(true, sql.contains(sql5));
+    Assert.assertEquals(true, sql.contains(sql6));
+
+    String transactionIdForSql2 =
+        requests.stream()
+            .filter(req -> req.getSql().equals(sql2))
+            .findAny()
+            .get()
+            .getTransaction()
+            .getId()
+            .toStringUtf8();
+    String transactionIdForSql3 =
+        requests.stream()
+            .filter(req -> req.getSql().equals(sql3))
+            .findAny()
+            .get()
+            .getTransaction()
+            .getId()
+            .toStringUtf8();
+
+    Assert.assertEquals(true, transactionIdForSql2.equals(transactionIdForSql3));
+
+    String transactionIdForSql5 =
+        requests.stream()
+            .filter(req -> req.getSql().equals(sql5))
+            .findAny()
+            .get()
+            .getTransaction()
+            .getId()
+            .toStringUtf8();
+    String transactionIdForSql6 =
+        requests.stream()
+            .filter(req -> req.getSql().equals(sql6))
+            .findAny()
+            .get()
+            .getTransaction()
+            .getId()
+            .toStringUtf8();
+
+    Assert.assertEquals(true, transactionIdForSql5.equals(transactionIdForSql6));
+  }
+
+  @Test
+  public void testAutocommitInTransactions() throws IOException, InterruptedException {
+    // tests autocommit settings using connection.autocommit variable
+    List<String> statements = new ArrayList<>();
+    String sql1 = "Select * from some_table";
+    String sql2 = "insert into some_table(col1, col2) values(value1, value2)";
+    String sql3 = "Select * from some_table3";
+    String sql4 = "insert into some_table3(col1, col2) values(value1, value2)";
+
+    // no commit request will be sent after this query because autocommit is false
+    statements.add("query");
+    statements.add(sql1);
+
+    // no commit request will be sent after this update because autocommit is false
+    statements.add("update");
+    statements.add(sql2);
+
+    // no commit request will be sent after this update because autocommit is false
+    statements.add("update");
+    statements.add(sql4);
+
+    // 1 commit request will be sent
+    statements.add("transaction");
+    statements.add("commit");
+
+    statements.add("transaction");
+    statements.add("set autocommit True");
+
+    // no commit request after this query although autocommit is true because it is a query sql
```

Comment:
Could we also add an assertion below that not only is there no commit request after this statement, but that it also uses a single-use read-only transaction? (So something like `assertTrue(request.getTransaction().hasSingleUse())` and `assertTrue(request.getTransaction().getSingleUse().hasReadOnly())`)